src/mondeath.cpp

```diff
@@ -405,7 +405,18 @@ void mdeath::worm( monster &z )
 
 void mdeath::disappear( monster &z )
 {
-    add_msg_if_player_sees( z, m_good, _( "The %s disappears." ), z.name() );
+
+    // The .compare() method returns 0 if equal
+    // Any other value is implicitly true
+    // so we need to reverse it
+    // substring(x,y) returns the y-character string starting at index x
+    bool begins_with_your = !( z.name().substr(0,4).compare("your") );
+    // this bool is now true if the enemy's name begins with 'your'
+
+    std::string msg = ( begins_with_your ) ?
+        _("%s disappears.") : _("The %s disappears.");
```

Comment:
If monster name begins with "your", then you may need to capitalise the first letter "y" since it's the first letter in the sentence now; however, this may not apply to other languages.

Reply:
Isn't that what `disp_name(false, true)` does?

Reply:
Sorry, you are right, I don't know that feature.